### PR TITLE
Delete project delete experiment [BIOMAGE-948]

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-948

#### Link to staging deployment URL 

In UI repo

#### Links to any Pull Requests related to this

UI : https://github.com/biomage-ltd/ui/pull/279
API : https://github.com/biomage-ltd/api/pull/131

#### Anything else the reviewers should know about the changes here

- Added new `deleteExepriment` action

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [X] Unit tests written
- [X] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [X] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/api/route-services/projects.js [hunk 1]
```diff
@@ -184,9 +184,11 @@ class ProjectsService {
       const { experiments } = await this.getProject(projectUuid);
 
       if (experiments.length > 0) {
```
Comment (ivababukova) [inline src/api/route-services/projects.js:188]: I saw in the logs that the line `experiments.length` sometimes throws an error, specifically when `experiments` is none -- it complains that there is not such thing as length of undefined. Will be good to add a check in the code to guard against those cases and return not found when `experiments` is undefined

**Comment by ivababukova** [comment]
One small thing, the rest looks good. After we check the staging link, let's merge

File: src/api/route-services/projects.js (hunk 1 above)
Comment (aerlaut) [reply src/api/route-services/projects.js:188]: Yea, it happens when we're loading projects for an experiment. This error always pop up if we switch to experiments without project (e.g. previously created experiments). I think it'll need fixes in the UI and API.

Comment (ivababukova) [reply src/api/route-services/projects.js:188]: What are the implications of this error? Is it just something annoying that shows up in logs all the time? What is its impact to the user?

Comment (ivababukova) [reply src/api/route-services/projects.js:188]: also, is it a small thing to fix? Do you think the fix makes sense to be in a separate ticket?

Comment (aerlaut) [reply src/api/route-services/projects.js:188]: Added checks around the call. This function is also called when checking credentials. There are still errors being shown in the UI, rendering errors for unrelated experiment. I'm not sure why, but it seems like the notification is "cached" or not entirely cleared.

**Review by ivababukova — APPROVED** [review]

File: src/api/route-services/projects.js (hunk 1 above)
Comment (aerlaut) [reply src/api/route-services/projects.js:188]: Moved into ticket https://biomage.atlassian.net/browse/BIOMAGE-1012

Comment (ivababukova) [reply src/api/route-services/projects.js:188]: thanks, let's look into this there

**Pull request merged**